### PR TITLE
Basic mobs now use z-level turnoff instead of simple

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

On one compile of MetaStation, I saw that there's 45 basic mobs on the station, 256 on lavaland (the number growing from tendrils), and 59 in all other z levels combined.

While we do expect Lavaland to be visited every round, at least it won't be running during the times when no one is there, but even more importantly, space exploration is something not done every round, so we don't have any reason to waste our resources on AIs that will never be interacted with.

Simple animals had an easy solution to this:
If no one is on the Z level, their AI turns off
If someone is on the Z level, they are idle unless needed.

The last simple animals that exists right now are bots, megafauna, geese, gondolas, and some minor ones like mimic, zombie, dark wizard, soulscythe, etc.
Point is, we're very much nearly done going through all simple animals, so this code is being wasted just to ensure things like cleanbots won't work if no one is on the z level, something I doubt happens often, so I took their code and made it work for basic mobs instead. I could've done both but I thought it would look very bad, and maybe this is a good incentivize to get more basic mob conversions.

There's one major change here and it's that we're missing the "Idle" mode, some basic mobs like the Lavaland village seems to be made with intent that they'll be running even if players aren't around, so this sets up a future PR that makes idle AI easier to add, and I want to make sure those cases are taken into account.

## Why It's Good For The Game

We don't need to always be processing these basic mobs, and sets us in the future to hopefully also implement idle AIs.

## Changelog

:cl:
balance: Basic mob AIs with no mobs on the Z level now stop.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/dead.dm [hunk 1]
```diff
@@ -90,16 +90,20 @@ INITIALIZE_IMMEDIATE(/mob/dead)
...
+/**
+ * updates the Z level for dead players
+ * If they don't have a new z, we'll keep the old one, preventing bugs from ghosting and re-entering, among others
+ */
+/mob/dead/proc/update_z(new_z)
+	if(registered_z == new_z)
+		return
+	if(registered_z)
+		SSmobs.dead_players_by_zlevel[registered_z] -= src
+	if(!client)
```
Comment (san7890) [inline code/modules/mob/dead/dead.dm:102]:
```suggestion
	if(isnull(client))
```

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -1748,32 +1748,30 @@ GLOBAL_LIST_EMPTY(fire_appearances)
...
-
-
-			registered_z = new_z
-		else
-			registered_z = null
+	if(registered_z == new_z)
+		return
+	if(registered_z)
+		SSmobs.clients_by_zlevel[registered_z] -= src
+	if(!client)
```
Comment (san7890) [inline code/modules/mob/living/living.dm:1755]:
```suggestion
	if(isnull(client))
```

**Review by san7890 — CHANGES_REQUESTED** [review]
thanks for doing this for basic mobs, just some nits

**Comment by LemonInTheDark** [comment]
based

**Comment by san7890** [comment]
this PR is absolutely killing CI with hour-long workflows and clogging up everything that requests runners

@JohnFulpWillard Please add [ci skip] to the title of your PR !!!AND!!! draft it as well. maybe even DNM. it's taking hours for our runners to free up every time you push a commit to remote (don't feel bad about this i do the same thing) but just FYIing you and giving a potential solution.

**Comment by san7890** [comment]
actually fuck my life it probably doesn't filter out `ci skip` since we don't check for that in the name of the PR. just uh cancel your workflows when you can then thanks

File: code/datums/ai/_ai_controller.dm [hunk 3]
```diff
@@ -156,11 +180,14 @@ multiple modular subtrees with behaviors
 	if(isnull(pawn))
 		return // instantiated without an applicable pawn, fine
 
-	UnregisterSignal(pawn, list(COMSIG_MOB_LOGIN, COMSIG_MOB_LOGOUT, COMSIG_MOB_STATCHANGE, COMSIG_QDELETING))
+	UnregisterSignal(pawn, list(COMSIG_MOVABLE_Z_CHANGED, COMSIG_MOB_LOGIN, COMSIG_MOB_LOGOUT, COMSIG_MOB_STATCHANGE, COMSIG_QDELETING))
 	if(ai_movement.moving_controllers[src])
 		ai_movement.stop_moving_towards(src)
 	pawn.ai_controller = null
 	pawn = null
+	var/turf/pawn_turf = get_turf(pawn)
```
Comment (Ben10Omintrix) [inline code/datums/ai/_ai_controller.dm:188]: pawn is nulled at this point

**Comment by JohnFulpWillard** [comment]
> this PR is absolutely killing CI with hour-long workflows and clogging up everything that requests runners
> 
> @JohnFulpWillard Please add [ci skip] to the title of your PR !!!AND!!! draft it as well. maybe even DNM. it's taking hours for our runners to free up every time you push a commit to remote (don't feel bad about this i do the same thing) but just FYIing you and giving a potential solution.
> 
> maybe another solution in my mind would be to stop CI from running on PRs in a draft state, but i think that invites more problems than solutions... anyways just to be courteous until the kinks are ironed out please just either do what i just mentioned or cancel workflows before it locks up everything.

sorry but isn't this the whole point of why ci exists

**Comment by san7890** [comment]
Yes but there's a finite supply of runners and several PRs that need it

**Comment by JohnFulpWillard** [comment]
i'll be back for this

File: code/controllers/subsystem/idlenpcpool.dm [hunk 4]
```diff
@@ -1,47 +0,0 @@
-SUBSYSTEM_DEF(idlenpcpool)
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/idlenpcpool.dm:1]: the corresponding ai flag was still referenced in simple mob code, it's only one spot but it does seem to be loadbearing, should not have  been removed

**Pull request merged**